### ISSUE TITLE
Memory error or something else?

### ISSUE THREAD

**Issue by kkakosim** [charlesll/Spectra.jl#16]
Hi,
I am fitting a baseline in a large dataset (52k points) with a memory error:
> ....julia\packages\PyCall\tqyST\src\pyfncall.jl:43 =# @pysym(:PyObject_Call), PyPtr, (PyPtr, PyPtr, PyPtr), o, pyargsptr, kw))))) <class 'MemoryError'>
> MemoryError((51752, 51752), dtype('float64'))...

the function call and arguments seem to be ok because it completes with way less data (~1k). Because of large gradients (sharp peaks) I would like to avoid decimating or smoothing the data. Is this a memory issue or something else?
(Julia v1.5.1, Win10: 16GB RAM)

**Comment by kkakosim** [comment]
to add on it.
the baseline function seems to be working fine when the same dataset is resampled to any of the tested sizes (1k-52k). There might be something wrong with the data, but then what is it with "MemoryError"

**Comment by charlesll** [comment]
This seems unrelated to rampy per se... Which type of baseline were you using when this happened?

**Comment by kkakosim** [comment]
I tried all of them. Currently, all but the `gcvspline` work after resampling the data (Something might have gone wrong during the `gcvspline` installation; later I will try with a fresh build/install) . 

The `MemoryError` is not an issue for my anymore, because of the resampling. I brought it forward because of the misleading message. If you have any ideas how to investigate it let me know. Alternatively, I can share the dataset and my implementation if you want to investigate it yourself.

**Comment by charlesll** [comment]
Yes, this would help because I don't understand the origin of this error. You can send that on my email maybe, lelosq@ipgp.fr

Thanks in advance !

**Comment by charlesll** [comment]
@kkakosim I never found the origin of this error. But the new version of Spectra (v2.0.1) does not rely on Python/Rampy. This should not be a problem anymore, update to this new version please. 

For now I am closing this issue. Re-open it if you encounter the issue again.

**Comment by kkakosim** [comment]
Thanks @charlesll for coming back to this and maintaining a wonderful OSS. I believe the issue was resolved in v1. I don't recall observing it again since that update.
